frontends/p4-14/p4-14-parse.ypp

```diff
@@ -824,31 +824,31 @@ expression:
     | LATEST '.' name   { $$ = new IR::Member(@1+@3, new IR::NamedRef(@1, $1), IR::ID(@3, $3)); }
     | INTEGER           { $$ = $1; }
     | '(' expression ')'                { $$ = $2; }
-    | NOT expression %prec PREFIX       { $$ = new IR::LNot(@1, $2); }
-    | '!' expression %prec PREFIX       { $$ = new IR::LNot(@1, $2); }
-    | '~' expression %prec PREFIX       { $$ = new IR::Cmpl(@1, $2); }
-    | '-' expression %prec PREFIX       { $$ = new IR::Neg(@1, $2); }
+    | NOT expression %prec PREFIX       { $$ = !$2; }
```

Comment:
The grammar was partly serving as documentation of the IR; now developers will have to read the ERef class too.

You are losing the sourceInfo. Error messages will not be able to point to the source code.